lib/tasks/scheduler.rake

```diff
@@ -0,0 +1,33 @@
+namespace :upcoming_lessons do
+
+  desc "This task checks to see if there are four upcoming lessons per matching. If not, add additional lessons"
+  task :verify_or_add => :environment do
+    puts "Verifying there exists four upcoming lesson for given matching..."
+
+    number_of_upcoming_lessons = 4
+    midnight =  Date.current.tomorrow.midnight
+
+    Matching.all.each do |matching|
+      upcoming_lessons = matching.lessons.where(is_paid: false).where("created_at >= :date", date: midnight)
+      if upcoming_lessons.count < number_of_upcoming_lessons
+        previous_lesson = matching.lessons.order("created_at DESC").first
+        create_single_lesson(matching, previous_lesson)
+      end
+    end
+    puts "Done."
+  end
+
+  def create_single_lesson(matching, previous_lesson)
+    start_time = previous_lesson.start_time + 7.day
+    end_time = previous_lesson.end_time + 7.day
```

Comment:
Matchings lesson time is an array of numbers... why is the lessons start_time and end_time datetime objects? Do I have to convert?